core/lib/zksync_core/src/lib.rs

```diff
@@ -102,7 +102,9 @@ pub async fn genesis_init(
     eth_sender: &ETHSenderConfig,
     network_config: &NetworkConfig,
     contracts_config: &ContractsConfig,
+    ethwatch_config: &ETHWatchConfig,
     eth_client_url: &str,
+    wait_for_set_chain_id: bool,
```

Comment:
what does this flag depend on? Will it be remived in the future, or will chains be able to spin up both ways?

Reply:
As per comment above, `zk server --genesis` will have to be run twice - with and without this flag.